spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshSupport.java

```diff
@@ -0,0 +1,206 @@
+package org.springframework.cloud.context.refresh;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.boot.Banner.Mode;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.cloud.context.environment.EnvironmentChangeEvent;
+import org.springframework.cloud.context.scope.refresh.RefreshScope;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.CompositePropertySource;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.EnumerablePropertySource;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.web.context.support.StandardServletEnvironment;
+
+/**
+ * 
+ * @author Venil Noronha
+ */
+public class RefreshSupport {
```

Comment:
Yeah, need a better name. We already have `EnvironmentManager`. What about `ContextRefresher`? I like the way that `ApplicationEvents` provide an extensibility already, though (you can listen for the event if you want to refresh more stuff). Maybe the `ContextRefresher` could just emit an event, and all the other features can be listeners?